File: Exercícios Aula 1-2-3/Vetor 1-4/func.py
import random

def fazLista(x):
    vetor = [0] * x

    for i in range(x):
        vetor[i] = int(input("Digite o valor da posicao {}: ".format(i)))

    return vetor

def fazListaAleatoria(x):
    vetor = []

    for i in range(x):
        vetor.append(int(random.uniform(0, 100)))

    return vetor

def somaListas(vetorA, vetorB):
    vetorC = []

    for i in range(len(vetorA)):
        vetorC.append(vetorA[i] + vetorB[i])

    return vetorC

def leitura():
    return int(input("Digite a posicao desejada: "))

def exibicao(vetor):
    for i in range(len(vetor)):
        print(vetor[i], " ", end=" ")
    print("")

def soma(x, y):
    return print(x + y)

def maiorEMenor(vetor):
    maior = 0
    menor = 99999

    for i in range(0, len(vetor)):
        if vetor[i] > maior:
            maior = vetor[i]
        if vetor[i] < menor:
            menor = vetor[i]
    return print("Maior elemento é ", maior ," e menor elemento e", menor)

def somaLista(vetor):
    soma = 0

    for i in range(0, len(vetor)):
        soma += vetor[i]
    return print("Soma e %d" % soma)

#  Esta função exibirá
#  • Os números pares digitados; 
#  • A soma dos numeros pares digitados; 
#  • Os números ímpares digitados;
#  • A quantidade de números ímpares
def parLista(vetor):
    numerosPares = []
    numerosImpares = []
    somaPares = 0

    for numero in vetor:
        if numero % 2 == 0:
            numerosPares.append(numero)
            somaPares += numero
        else:
            numerosImpares.append(numero)
    
    print("Os números pares digitados são: ", end="")
    for numero in numerosPares:
        print(numero, " ", end="")

    print("\nA soma dos números pares digitados é:", somaPares,)

    print("Os números ímpares digitados são: ", end="")
    for numero in numerosImpares:
        print(numero, " ", end="")

    print("\nA quantidade de números ímpares digitados é:", len(numerosImpares))

File: Exercícios Aula 1-2-3/Vetor 1-4/Vetor em java/src/Func.java
import java.util.ArrayList;
import java.util.Random;
import java.util.Scanner;
public class Func {
    public static ArrayList<Integer> faz_lista(int x){
        ArrayList<Integer> vetor = new ArrayList<>();
        Scanner scan = new Scanner(System.in);

        for(int i = 0; i < x; i++){
            System.out.println("Digite o valor da posição " + i + ": ");
            vetor.add(scan.nextInt());
        }
        return vetor;
    }
    public static ArrayList<Integer> faz_lista_aleatoria(int x){
        ArrayList<Integer> vetor = new ArrayList<>();
        Random rand = new Random();

        for(int i = 0; i < x; i++){
            vetor.add(rand.nextInt(0, 100));
        }
        return vetor;
    }
    public static ArrayList<Integer> soma_listas(ArrayList<Integer> vetorA, ArrayList<Integer> vetorB){
        ArrayList<Integer> vetorC = new ArrayList<>();

        for(int i = 0; i < vetorA.size(); i++){
            vetorC.add(vetorA.get(i) + vetorB.get(i));
        }

        return vetorC;
    }
    public static int leitura(){
        Scanner scan = new Scanner(System.in);
        System.out.println("Digite o valor da posição desejada: ");
        return(scan.nextInt());
    }

    public static void exibicao(ArrayList<Integer> vetor){
        for(int i = 0; i < vetor.size(); i++){
            System.out.print(vetor.get(i) + " ");

        }
        System.out.println();
    }
    public static void soma(int x, int y){
        System.out.println(x + y);
    }
    public static void maior_menor(ArrayList<Integer> vetor){
        int maior = 0;
        int menor = 9999;

        for(int i = 0; i < vetor.size(); i++){
            if(vetor.get(i) > maior){
                maior = vetor.get(i);
            }
            if(vetor.get(i) < menor){
                menor = vetor.get(i);
            }
        }
        System.out.println("Maior elemento é " + maior + " e o menor é " + menor);
    }
    public static void soma_lista(ArrayList<Integer> vetor){
        int soma = 0;
        for(int i : vetor){
            soma += i;
        }
        System.out.println("Soma dos valores é " + soma);
    }
    /*  Esta função exibirá
        • Os números pares digitados;
        • A soma dos numeros pares digitados;
        • Os números ímpares digitados;
        • A quantidade de números ímpares */
    public static void par_impar_lista(ArrayList<Integer> vetor){
        ArrayList<Integer> pares = new ArrayList<>();
        ArrayList<Integer> impares = new ArrayList<>();
        int soma_pares = 0;

        for(int i : vetor){
            if(i % 2 == 0){
                pares.add(i);
                soma_pares += i;
            }else{
                impares.add(i);
            }
        }

        System.out.print("Os números pares digitatos são: ");
        for(int i : pares){
            System.out.print(i + " ");
        }
        System.out.println();
        System.out.println("A soma dos números pares é: " + soma_pares);

        System.out.print("Os números ímpares digitatos são: ");
        for(int i : impares){
            System.out.print(i + " ");
        }
        System.out.println();
        System.out.println("A quantidade de números ímpares digitados é: " + impares.size());
    }
}
